internal/infrastructure/kubernetes/deployment.go

```diff
@@ -250,13 +252,16 @@ func (i *Infra) createDeployment(ctx context.Context, infra *ir.Infra) (*appsv1.
 //
 // TODO: Update k/v pair to use gatewayclass controller name to distinguish between
 //       multiple Envoy Gateways.
-func EnvoyPodSelector() *metav1.LabelSelector {
+func EnvoyPodSelector(gatewayName string) *metav1.LabelSelector {
```

Comment:
The input name should be `className`, `gcName`, or something that represents a GatewayClass and not Gateway b/c the ProxyInfra name is established from the managed GatewayClass.

xref: https://github.com/envoyproxy/gateway/blob/v0.2.0-rc1/internal/gatewayapi/translator.go#L97

Reply:
Yes, a GatewayClass is not 1:1 with a Gateway. Envoy Gateway supports multiple Gateways but only one GatewayClass that matches its configured controller name. Therefore, there is a 1:1 mapping of managed proxy infra with a GatewayClass. Envoy Gateway manages a single data plane, e.g. the Envoy Deployment. From a Kubernetes Infra Manager perspective, Gateways correlate to a Service. Therefore, it does make sense for Services to include a GatewayClass __and__ a Gateway label. This will allow users to understand what Gateway was used to create the Service. The selector for the Envoy Service should continue using the pod labels of the Envoy Deployment. [This section ](https://github.com/envoyproxy/gateway/blob/main/docs/design/SYSTEM_DESIGN.md#design-decisions)of the design doc provides additional background.

Reply:
so, we should put both GatewayClass and Gateway's name?

Reply:
@zirain I suggest we keep it simple and just use the GatewayCLass name for now.

Reply:
updated
BTW, what's the plan about support multi gateway?